mne/minimum_norm/psf_ctf.py

```diff
@@ -0,0 +1,928 @@
+    mode : 'mean' | 'sum' | 'svd' |
+        PSFs can be computed for different summary measures with labels:
+        'sum' or 'mean': sum or means of sub-leadfields for labels
+        This corresponds to situations where labels can be assumed to be
+        homogeneously activated.
+        'svd': SVD components of sub-leadfields for labels
+        This is better suited for situations where activation patterns are
+        assumed to be more variable.
+        "sub-leadfields" are the parts of the forward solutions that belong to
+        vertices within invidual labels.
+    n_svd_comp : integer
+        Number of SVD components for which PSFs will be computed and output
+        (irrelevant for 'sum' and 'mean'). Explained variances within
+        sub-leadfields are shown in screen output.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    stc_psf : SourceEstimate
+        The PSFs for the specified labels
+        If mode='svd': n_svd_comp components per label are created
+        (i.e. n_svd_comp successive time points in mne_analyze)
+        The last sample is the summed PSF across all labels
+        Scaling of PSFs is arbitrary, and may differ greatly among methods
+        (especially for MNE compared to noise-normalized estimates).
+    label_singvals : list of dict
+        Singular values of SVD for sub-matrices of leadfield
+        (only if mode='svd').
+        Diffent list entries per label. Since SVD is applied separately for
+        channel types, dictionaries may contain keys 'mag', 'grad' or 'eeg'.
+    evoked_fwd : Evoked
+        Forward solutions corresponding to PSFs in stc_psf
+        If mode='svd': n_svd_comp components per label are created
+        (i.e. n_svd_comp successive time points in mne_analyze)
+        The last sample is the summed forward solution across all labels
+        (sum is taken across summary measures).
+    """
+
+<<<<<<< HEAD
+    print "Point Spread Function!"
+
+=======
+>>>>>>> 909407689498cb58d8026f18798a8dd210d4f256
+    mode = mode.lower()
+    if mode not in ['mean', 'sum', 'svd']:
+        raise ValueError("mode must be 'svd', 'mean' or 'sum'. Got %s."
+                         % mode)
+
+    logger.info("About to process %d labels" % len(labels))
+
+    # CHANGED (PSF and CTF differ with respect to source orientations?)
+    # for forward computations, orientations will be fixed
+    forward = convert_forward_solution(forward, force_fixed=True,
+                                       surf_ori=True)
+
+    info_inv = _prepare_info(inverse_operator)
+
+    fwd_ch_names = forward['info']['ch_names']
+    inv_ch_names = info_inv['ch_names']
+<<<<<<< HEAD
+    ch_names = [c for c in inv_ch_names if (c not in info_inv['bads'])]
+=======
+    ch_names = [c for c in inv_ch_names if (c not in info_inv['bads']) ]
+>>>>>>> 909407689498cb58d8026f18798a8dd210d4f256
```

Comment:
and another one. look for those guys. Not sure how difficult it is. maybe start all over again and copy things in clean file and one commit